packages/yarnpkg-nm/sources/buildNodeModulesTree.ts

```diff
@@ -463,20 +463,16 @@ function getTargetLocatorPath(locator: PhysicalPackageLocator, pnp: PnpApi, opti
   if (info === null)
     throw new Error(`Assertion failed: Expected the package to be registered`);
 
-  let linkType;
-  let target;
-  if (options.pnpifyFs) {
-    // In case of pnpifyFs we represent modules as symlinks to archives in NodeModulesFS
-    // `/home/user/project/foo` is a symlink to `/home/user/project/.yarn/.cache/foo.zip/node_modules/foo`
-    // To make this fs layout work with legacy tools we make
-    // `/home/user/project/.yarn/.cache/foo.zip/node_modules/foo/node_modules` (which normally does not exist inside archive) a symlink to:
-    // `/home/user/project/node_modules/foo/node_modules`, so that the tools were able to access it
-    target = npath.toPortablePath(info.packageLocation);
-    linkType = LinkType.SOFT;
-  } else {
-    target = getRealPackageLocation(info, locator, pnp);
-    linkType = info.linkType;
-  }
+  // In case of pnpifyFs we represent modules as symlinks to archives in NodeModulesFS
```

Comment:
I think this change makes the code less readable compared to its current version. Stricter and unambigous explanation of intent is more important than const vs let.

Reply:
I think no. Though you can keep the `linkType` and `target` together and still do your refactoring, for example via;
```
  const {linkType, target} = options.pnpifyFs
    // Comment goes here
    ? {linkType: LinkType.SOFT, target: npath.toPortablePath(info.packageLocation)}
    : {linkType: info.linkType, target: getRealPackageLocation(info, locator, pnp)};
```
or even with:
```
  const [linkType, target] = options.pnpifyFs
    // Comment goes here
    ? [LinkType.SOFT, npath.toPortablePath(info.packageLocation)]
    : [info.linkType, getRealPackageLocation(info, locator, pnp)];
```

Reply:
Done in [`86b19a8` (#4143)](https://github.com/yarnpkg/berry/pull/4143/commits/86b19a80566f5805f448f9b7436dc2507f0521c5)